packages/code-du-travail-api/src/server/routes/search/getSearch.js

```diff
@@ -29,14 +29,18 @@ const getSearch = async querystring => {
     return knownQueryResult;
   }
   // remove console.log when stavble
-  logger.info(`querying sem search on: ${NLP_URL}/api/search?q=${query}`);
+  logger.info(
+    `querying sem search on: ${NLP_URL}/api/search?q=${encodeURIComponent(
+      query
+    )}`
+  );
 
   // we filter results to remove snippet document from main results
   const size = Math.min(querystring.size || MAX_RESULTS, 100);
 
-  const query_vector = await fetch(`${NLP_URL}/api/search?q=${query}`).then(
-    response => (response = response.json())
-  );
+  const query_vector = await fetch(
+    `${NLP_URL}/api/search?q=${encodeURIComponent(query)}`
```

Comment:
@nkrmr Je crois que ça posait un soucis avec les concordances de résultats ça non ?

Reply:
comment ça ? sinon il faut vérifier ce qu'on reçoit coté nlp

Reply:
le "?" non encodé ca marche pas :)

Reply:
Le encodeURIComponent effectivement pose problème sur certaine requêtes

Reply:
notamment avec les apostrophes

Reply:
Le problème [existe aussi](http://master.code-du-travail-numerique.dev.factory.social.gouv.fr/modeles-de-courriers/releve-dheures-supplementaires?q=l%27heure) en effet avec les apostrophes. Cette PR fixe ce cas aussi

rule of thumb : always encode your url parameters :)

Reply:
Je crois que ce que Nico veut dire c'est que si on fait un encoreURI sur une requete avec un apostrophe, on a des résultats pas cohérents avec ceux de la search. Enfin ceci dit il va falloir trouver une autre solution parce que tout est cassé là vi

Reply:
@UnbearableBear @nkrmr si vous avez un exemple de requête qui était différent avec `'` ou `?`

Reply:
c'est à dire?